test/jdk/jdk/internal/platform/cgroup/TestCgroupSubsystemFactory.java

```diff
@@ -340,6 +361,20 @@ public void testHybridCgroupsV1() throws IOException {
         assertEquals("/sys/fs/cgroup/memory", memoryInfo.getMountPoint());
     }
 
+    @Test
+    public void testColonsCgroupsV1() throws IOException {
+        String cgroups = cgroupv1CgInfoNonZeroHierarchy.toString();
+        String mountInfo = cgroupv1MntInfoNonZeroHierarchy.toString();
+        String selfCgroup = cgroupv1SelfColons.toString();
+        Optional<CgroupTypeResult> result = CgroupSubsystemFactory.determineType(mountInfo, cgroups, selfCgroup);
+
+        assertTrue("Expected non-empty cgroup result", result.isPresent());
+        CgroupTypeResult res = result.get();
+        CgroupInfo memoryInfo = res.getInfos().get("memory");
+        assertEquals(memoryInfo.getCgroupPath(), "/system.slice/containerd.service/kubepods-burstable-podf65e797d_d5f9_4604_9773_94f4bb9946a0.slice:cri-containerd:86ac6260f9f8a9c1276748250f330ae9c2fcefe5ae809364ad1e45f3edf7e08a");
+        assertEquals(memoryInfo.getMountRoot(), memoryInfo.getMountRoot());
```

Comment:
What I meant here is to also change the mountinfo entry for `memory` and then use
```
assertEquals(memoryInfo.getMountRoot(), memoryInfo.getCgroupPath());
```
as that's really what was happening in the wild. mount root - containing colons - would correctly parse, but the cgroup path - containing colons - would not and the assertion would fail (without the change in CgroupSubsystemFactory). Thanks!

Reply:
@hseigel it still has `assertEquals(memoryInfo.getMountRoot(), memoryInfo.getMountRoot());` - tautology - should be: `assertEquals(memoryInfo.getMountRoot(), memoryInfo.getCgroupPath());`